{
  "guild": {
    "id": "484437221055922177",
    "name": "Mina Protocol",
    "iconUrl": "https://cdn.discordapp.com/icons/484437221055922177/60f5907d2f7e3804b704a7153cae7768.png?size=512"
  },
  "channel": {
    "id": "1129663134660050994",
    "type": "GuildPublicThread",
    "categoryId": "1047214314349658172",
    "category": "zkapps-questions",
    "name": "Atomics.wait cannot be called in this context",
    "topic": null
  },
  "dateRange": {
    "after": null,
    "before": null
  },
  "exportedAt": "2025-04-24T14:57:27.6136158+04:00",
  "messages": [
    {
      "id": "1129663134660050994",
      "type": "Default",
      "timestamp": "2023-07-15T10:37:59.429+04:00",
      "timestampEdited": null,
      "callEndedTimestamp": null,
      "isPinned": false,
      "content": "I am getting this error when I try to compile the zkApp in the UI.\n```\n    let zkProgram = (await import('game_of_life')).zkProgram;\n    let contract = (await import('game_of_life')).contract;\n    await zkProgram.compile(); // throws error\n    await contract.compile();\n    . . .\n```\nHow to resolve this?",
      "author": {
        "id": "368163996219604992",
        "name": "thedude_og",
        "discriminator": "0000",
        "nickname": "TheDude",
        "color": "#FA7504",
        "isBot": false,
        "roles": [
          {
            "id": "1182371055923376168",
            "name": "Mina Navigator",
            "color": "#FA7504",
            "position": 100
          },
          {
            "id": "1150768164850913322",
            "name": "zkIgnite Cohort 2",
            "color": "#E97404",
            "position": 96
          },
          {
            "id": "1220064445863038996",
            "name": "zkApps",
            "color": "#FFFFFF",
            "position": 19
          }
        ],
        "avatarUrl": "https://cdn.discordapp.com/embed/avatars/1.png"
      },
      "attachments": [],
      "embeds": [],
      "stickers": [],
      "reactions": [
        {
          "emoji": {
            "id": "",
            "name": "\uD83D\uDC4D",
            "code": "thumbsup",
            "isAnimated": false,
            "imageUrl": "https://cdn.jsdelivr.net/gh/twitter/twemoji@latest/assets/svg/1f44d.svg"
          },
          "count": 2,
          "users": [
            {
              "id": "301752126491721730",
              "name": "blockbeard",
              "discriminator": "0000",
              "nickname": "angus_m",
              "isBot": false,
              "avatarUrl": "https://cdn.discordapp.com/avatars/301752126491721730/0a8d7f37a46cb3c2ca757f772e60e497.png?size=512"
            },
            {
              "id": "915265080210366544",
              "name": "sniping6882",
              "discriminator": "0000",
              "nickname": "Sniping",
              "isBot": false,
              "avatarUrl": "https://cdn.discordapp.com/avatars/915265080210366544/a034ecb9015241bd18c016f5e539908f.png?size=512"
            }
          ]
        }
      ],
      "mentions": []
    },
    {
      "id": "1130000438394097684",
      "type": "Default",
      "timestamp": "2023-07-16T08:58:18.906+04:00",
      "timestampEdited": null,
      "callEndedTimestamp": null,
      "isPinned": false,
      "content": "I think it's an issue with versions\nI upgraded snarkyjs to  `0.12.1` \nand now I am getting a different error\n```\n1. We can't find this zkapp account in the ledger\n2. Because the zkapp account was not found in the cache. Try calling `await fetchAccount(zkappAddress)` first.\nIf none of these are the case, then please reach out on Discord at #zkapp-developers and/or open an issue to tell us!\n```\nadding `await fetchAccount(zkappAddress)` does not help",
      "author": {
        "id": "368163996219604992",
        "name": "thedude_og",
        "discriminator": "0000",
        "nickname": "TheDude",
        "color": "#FA7504",
        "isBot": false,
        "roles": [
          {
            "id": "1182371055923376168",
            "name": "Mina Navigator",
            "color": "#FA7504",
            "position": 100
          },
          {
            "id": "1150768164850913322",
            "name": "zkIgnite Cohort 2",
            "color": "#E97404",
            "position": 96
          },
          {
            "id": "1220064445863038996",
            "name": "zkApps",
            "color": "#FFFFFF",
            "position": 19
          }
        ],
        "avatarUrl": "https://cdn.discordapp.com/embed/avatars/1.png"
      },
      "attachments": [],
      "embeds": [],
      "stickers": [],
      "reactions": [],
      "mentions": []
    },
    {
      "id": "1130018313628549191",
      "type": "Reply",
      "timestamp": "2023-07-16T10:09:20.694+04:00",
      "timestampEdited": null,
      "callEndedTimestamp": null,
      "isPinned": false,
      "content": "It turns out I was missing\n```\n  Mina.Network('https://proxy.berkeley.minaexplorer.com/graphql');\n```\nThe error msg was no help.",
      "author": {
        "id": "368163996219604992",
        "name": "thedude_og",
        "discriminator": "0000",
        "nickname": "TheDude",
        "color": "#FA7504",
        "isBot": false,
        "roles": [
          {
            "id": "1182371055923376168",
            "name": "Mina Navigator",
            "color": "#FA7504",
            "position": 100
          },
          {
            "id": "1150768164850913322",
            "name": "zkIgnite Cohort 2",
            "color": "#E97404",
            "position": 96
          },
          {
            "id": "1220064445863038996",
            "name": "zkApps",
            "color": "#FFFFFF",
            "position": 19
          }
        ],
        "avatarUrl": "https://cdn.discordapp.com/embed/avatars/1.png"
      },
      "attachments": [],
      "embeds": [],
      "stickers": [],
      "reactions": [
        {
          "emoji": {
            "id": "",
            "name": "\uD83D\uDD25",
            "code": "fire",
            "isAnimated": false,
            "imageUrl": "https://cdn.jsdelivr.net/gh/twitter/twemoji@latest/assets/svg/1f525.svg"
          },
          "count": 1,
          "users": [
            {
              "id": "511272055623122961",
              "name": "laurip",
              "discriminator": "0000",
              "nickname": "LauriP",
              "isBot": false,
              "avatarUrl": "https://cdn.discordapp.com/avatars/511272055623122961/d57c3968960fb6a823b75cbd5530d6cf.png?size=512"
            }
          ]
        }
      ],
      "mentions": [],
      "reference": {
        "messageId": "1130000438394097684",
        "channelId": "1129663134660050994",
        "guildId": "484437221055922177"
      }
    }
  ],
  "messageCount": 3
}